### PR TITLE
feat: use `for(;;)` style loops for some `for [a..b]` style loops

### PR DESCRIPTION
**Pull request by eventualbuddha**
This commit does not use `for(;;)` when the loop has a dynamic range unless it also has an explicit step. That means `for [a..b]` will become a `for…of` loop. I added two skipped tests for turning them into `for(;;)` style, but it would require a fairly verbose loop header. However, that may still be better than requiring the `__range__` helper.

### CODE REVIEW

**Comment by eventualbuddha** [comment]
I believe this will fix #322.

File: test/for_test.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import check from './support/check.js';
 import validate from './support/validate.js';
 
-describe('for loops', () => {
+describe.only('for loops', () => {
```
Comment (alangpierce) [inline test/for_test.js]: Looks like you left the `only` in by mistake.

**Comment by alangpierce** [comment]
Looks good to me, although you accidentally disabled all of the other tests. :-)

Kind of sad that we can't do this transform for dynamic ranges. I have a lot of code like `for i of [0...len]`, where a human could obviously see that it means to count up. Maybe there could be a loose mode that has a heuristic like "if the range starts at the constant 0, patch as if the loop is counting up". Fortunately, this change makes it pretty easy to add those types of optional heuristics if we want. Or I could just go through code beforehand and write `by 1` when I'm confident that it's counting up.

I feel like I would pretty much never write code that decides at runtime whether to count up or down, so I maybe even prefer `__range__` over CoffeeScript's output because it makes it more obvious that someone should go in and write it the right way.

File: test/for_test.js (hunk 1 above)
Comment (eventualbuddha) [inline test/for_test.js]: Eek! Fixed.

**Pull request merged**